src/controller.rs

```diff
@@ -210,12 +211,16 @@ fn apply_mappings(
                 let source_metadata = convert_metadata(&subtree["metadata"]);
                 let mut subtree = subtree.clone();
                 cleanup_subtree(&mut subtree);
-                let mut source = DynamicObject::new(&subtree["metadata"]["name"].to_string(), &ar)
-                    .within(&subtree["metadata"]["namespace"].to_string())
-                    .data(subtree);
+                let mut source = if let Value::Null = subtree["metadata"]["namespace"] {
+                    DynamicObject::new(&subtree["metadata"]["name"].to_string(), &ar).data(subtree)
+                } else {
+                    DynamicObject::new(&subtree["metadata"]["name"].to_string(), &ar)
+                        .within(&subtree["metadata"]["namespace"].to_string())
+                        .data(subtree)
+                };
```

Comment:
similar situation here

```suggestion
                let mut source = DynamicObject::new(&subtree["metadata"]["name"].to_string(), &ar).data(subtree);
                source.metadata.namespace = subtree["metadata"]["namespace"].as_str().map(|s| s.to_string());
```